include/rocksdb/statistics.h

```diff
@@ -470,11 +470,18 @@ class Statistics {
     return type < HISTOGRAM_ENUM_MAX;
   }
 
+  virtual double getHistogramPercentile(uint32_t type,
+                                        double percentile) const = 0;
+  virtual uint64_t getHistogramSum(uint32_t type) const = 0;
+
   StatsLevel stats_level_ = kExceptDetailedTimers;
 };
 
 // Create a concrete DBStatistics object
-std::shared_ptr<Statistics> CreateDBStatistics();
+// Set use_histogram_windowing to true to use histogram_windowing
+// to report histogram stats.
+std::shared_ptr<Statistics> CreateDBStatistics(bool use_histogram_windowing =
```

Comment:
Do we need a way to pass parameters to `HistogramWindowingImpl`'s constructor? It looks like this always calls default constructor, which makes the window five minutes.

Reply:
To support passing windowing parameters through CreateDBStatistics, I think we need to use aligned storage and placement new constructors for both HistogramWindowingImpl[] in StatisticsData and for StatisticsData constructors in CoreLocalArray. Also, windowing parameters are not scenario based and would not change from use case to use case. They define the granularity at the rocksdb library level. The default parameters in the histogram_windowing header could be modified to use different window size.